build_all.sh

```diff
@@ -10,14 +11,27 @@ echo_run(){
   "$@"
 }
 
-[ -d csources ] || echo_run git clone --depth 1 https://github.com/nim-lang/csources.git
+[ -d csources ] || echo_run git clone -q --depth 1 https://github.com/nim-lang/csources.git
 
 nim_csources=bin/nim_csources
 build_nim_csources(){
   ## avoid changing dir in case of failure
   (
-    echo_run cd csources
-    echo_run sh build.sh $@
+    if [[ $# -ne 0 ]]; then
+      # some args were passed (eg: `--cpu i386`), need to call build.sh
+      echo_run cd csources
+      echo_run sh build.sh $@
+    else
+      # no args, use multhreaded (5X faster on 16 cores: 10s instead of 50s)
+      makeX=make
+      unamestr=$(uname)
+      if [ "$unamestr" = 'FreeBSD' ]; then
+        makeX=gmake
+      fi
+      logicalCpus=$([ $(uname) = 'Darwin' ] && sysctl -n hw.logicalcpu_max || nproc)
```

Comment:
NIT: `getconf _NPROCESSORS_ONLN` works on both macOS and Linux, but I'm not sure works on FreeBSD or else (maybe it requires coreutils).

Reply:
I found thist: https://stackoverflow.com/questions/45181115/portable-way-to-find-the-number-of-processors-cpus-in-a-shell-script

Reply:
ok, now using this: https://stackoverflow.com/a/53427092/1426932 but with a slight modification because of https://gitlab.haskell.org/ghc/ghc/issues/8594